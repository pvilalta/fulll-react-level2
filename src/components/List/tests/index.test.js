import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import Component from '..';

describe('components | List | index', () => {
  it('should render component', () => {
    const items = [
      {
        login: 'frodon',
        id: 479213,
      },
      {
        login: 'sam',
        id: 479214,
      },
    ];

    const { container } = render(<Component items={items} />);
    const profileOne = screen.getByText('frodon');
    const profileTwo = screen.getByText('sam');

    expect(container).toMatchSnapshot();
    expect(profileOne).toBeInTheDocument();
    expect(profileTwo).toBeInTheDocument();
  });
});
