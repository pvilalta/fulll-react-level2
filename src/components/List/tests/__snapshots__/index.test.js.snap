// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | List | index should render component 1`] = `
<div>
  <div
    class="List"
  >
    <a
      rel="noreferrer"
      target="_blank"
    >
      <div
        class="user"
      >
        <div
          class="userImage"
        >
          <img
            alt="avatar of frodon"
          />
        </div>
        <div
          class="userInfo"
        >
          <span
            class="userName"
          >
            frodon
          </span>
          <span
            class="userID"
          >
            479213
          </span>
        </div>
      </div>
    </a>
    <a
      rel="noreferrer"
      target="_blank"
    >
      <div
        class="user"
      >
        <div
          class="userImage"
        >
          <img
            alt="avatar of sam"
          />
        </div>
        <div
          class="userInfo"
        >
          <span
            class="userName"
          >
            sam
          </span>
          <span
            class="userID"
          >
            479214
          </span>
        </div>
      </div>
    </a>
  </div>
</div>
`;
